modules/table/src/integrationTest/java/org/apache/ignite/internal/disaster/ItDisasterRecoveryReconfigurationTest.java

```diff
@@ -1542,12 +1851,25 @@ private List<Integer> getRealAssignments(IgniteImpl node0, int partId) {
         return stable.empty() ? null : Assignments.fromBytes(stable.value());
     }
 
+    private @Nullable AssignmentsChain getAssignmentsChain(IgniteImpl node, int partId) {
+        CompletableFuture<Entry> chainFut = node.metaStorageManager()
+                .get(assignmentsChainKey(new TablePartitionId(tableId, partId)));
+
+        assertThat(chainFut, willCompleteSuccessfully());
+
+        Entry chain = chainFut.join();
+
+        return chain.empty() ? null : AssignmentsChain.fromBytes(chain.value());
+    }
+
     @Retention(RetentionPolicy.RUNTIME)
     @interface ZoneParams {
         int replicas();
 
         int partitions();
 
         int nodes() default INITIAL_NODES;
+
+        ConsistencyMode consistencyMode() default ConsistencyMode.STRONG_CONSISTENCY;
     }
 }
```

Comment:
I'd say that we miss:

1. Verification that graceful change will rewrite chain after force resets.
2. Verification that !HA zone has empty chain.